File: problems/Longest Palindromic Substring/longestPalindrome.cs
public class Solution {
    public string LongestPalindrome(string s) {
        return longestPalindromeBottomUp(s);
    }

    private string longestPalindromeBottomUp(string s) {
        int n = s.Length;
        bool[,] dp = new bool[n, n];
        string lp = "";

        for (int left = n - 1; 0 <= left; --left) {
            for (int right = left; n > right; ++right) {
                if (s[left] == s[right]) {
                    if (2 >= 1 + right - left || dp[1 + left, right - 1]) {
                        dp[left, right] = true;

                        if (lp.Length < 1 + right - left) {
                            lp = s.Substring(left, 1 + right - left);
                        }
                    }
                }
            }
        }

        return lp;
    }
}

File: problems/Longest Palindromic Substring/longestPalindrome.js
/**
 * @param {string} s
 * @return {string}
 */
var longestPalindrome = function(s) {
    if (!s || !s.length) {
        return '';
    }

    const expandAroundCenter = function(s, left, right) {
        while (0 <= left && s.length > right && s.charAt(left) === s.charAt(right)) {
            --left;
            ++right;
        }

        return right - left - 1;
    }

    let start = 0;
    let end = 0;

    for (let idx = 0; s.length > idx; ++idx) {
        let len1 = expandAroundCenter(s, idx, idx);
        let len2 = expandAroundCenter(s, idx, 1 + idx);
        let len = Math.max(len1, len2);

        if (end - start < len) {
            start = idx - Math.trunc((len - 1) / 2);
            end = idx + Math.trunc(len / 2);
        }
    }

    return s.substring(start, 1 + end);
};
